From: Heinrich Schuchardt <heinrich.schuchardt@canonical.com>
Date: Mon, 9 Dec 2024 12:41:10 +0100
Subject: virt-aa-helper: allow riscv64 EDK II

Debian has packaged EDK II for 64-bit RISC-V in directory
/usr/share/qemu-efi-riscv64/.

For usage with libvirt update the AppArmor helper.

Signed-off-by: Heinrich Schuchardt <heinrich.schuchardt@canonical.com>
Reviewed-by: Andrea Bolognani <abologna@redhat.com>
(cherry picked from commit c63bdd17b9f5c31a2511f173d60455b83d22c561)

Forwarded: not-needed
Origin: https://gitlab.com/libvirt/libvirt/-/commit/c63bdd17b9f5c31a2511f173d60455b83d22c561
---
 src/security/virt-aa-helper.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/security/virt-aa-helper.c b/src/security/virt-aa-helper.c
index e82b5de..1cf9d7a 100644
--- a/src/security/virt-aa-helper.c
+++ b/src/security/virt-aa-helper.c
@@ -481,6 +481,7 @@ valid_path(const char *path, const bool readonly)
         "/usr/share/AAVMF/",
         "/usr/share/qemu-efi/",              /* for AAVMF images */
         "/usr/share/qemu-efi-aarch64/",
+        "/usr/share/qemu-efi-riscv64/",
         "/usr/share/qemu/",                  /* SUSE path for OVMF and AAVMF images */
         "/usr/lib/u-boot/",
         "/usr/lib/riscv64-linux-gnu/opensbi",
